pytesmo/validation_framework/adapters.py

```diff
@@ -34,9 +35,39 @@
 import operator
 from pytesmo.time_series.anomaly import calc_anomaly
 from pytesmo.time_series.anomaly import calc_climatology
+from pandas import DataFrame
 
 
-class MaskingAdapter(object):
+class BasicAdapter(object):
+    """
+    Base class for other adapters that works around the ascat reader not
+    returning a DataFrame. Also removes unnecessary timezone information in data.
+    """
+    def __init__(self, cls):
+        self.cls = cls
+
+    def __get_dataframe(self, data):
+        if ((not isinstance(data, DataFrame)) and (hasattr(data, 'data')) and (isinstance(data.data, DataFrame))):
+            data = data.data
```

Comment:
could add a key word argument for the name of the 'data', i.e. (hasattr(data, keyword_name_of_data)). Then if not named 'data', this could be overwritten. May be useful if a reader returns more than one data frame with data in it.